src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,17 +1,34 @@
+import { actions as TodoActions } from '../../features/filter';
+import { useAppDispatch } from '../../features/hooks';
+import { FilterStatus } from '../../utils/FilterStatus';
 import React from 'react';
 
-export const TodoFilter: React.FC = () => {
+type Props = {
+  filterBy: string;
+  query: string;
+};
+
+export const TodoFilter: React.FC<Props> = ({ query, filterBy }) => {
+  const dispatch = useAppDispatch();
+
+  const setFilterBy = (status: FilterStatus) =>
+    dispatch(TodoActions.setFilter(status));
+
+  const setQuery = (queryToSet: string) =>
+    dispatch(TodoActions.setQuery(queryToSet));
+
   return (
-    <form
-      className="field has-addons"
-      onSubmit={event => event.preventDefault()}
-    >
+    <form className="field has-addons">
       <p className="control">
         <span className="select">
-          <select data-cy="statusSelect">
-            <option value="all">All</option>
-            <option value="active">Active</option>
-            <option value="completed">Completed</option>
+          <select
+            data-cy="statusSelect"
+            value={filterBy}
+            onChange={event => setFilterBy(event.target.value as FilterStatus)}
```

Comment:
The type casting `event.target.value as FilterStatus` assumes that the value will always match a `FilterStatus` type. Ensure that the values in the select options correspond exactly to the `FilterStatus` enum values to prevent runtime errors.